File: tests/bonus/fixtures/hello.php
#!/usr/bin/env php
<?php
/**
 * Minimal CGI script that echoes REQUEST_METHOD and QUERY_STRING.
 */
$method = isset($_SERVER["REQUEST_METHOD"]) ? $_SERVER["REQUEST_METHOD"] : "UNKNOWN";
$query = isset($_SERVER["QUERY_STRING"]) ? $_SERVER["QUERY_STRING"] : "";
$cookie = isset($_SERVER["HTTP_COOKIE"]) ? $_SERVER["HTTP_COOKIE"] : "";

$body = "method=" . $method . "\nquery=" . $query . "\ncookie=" . $cookie . "\n";

header("Content-Type: text/plain");
header("Content-Length: " . strlen($body));
echo $body;
?>

File: tests/bonus/fixtures/hello.py
#!/usr/bin/env python3
"""
Minimal CGI script that echoes REQUEST_METHOD and QUERY_STRING.
"""
import os
import sys

method = os.environ.get("REQUEST_METHOD", "UNKNOWN")
query = os.environ.get("QUERY_STRING", "")
cookie = os.environ.get("HTTP_COOKIE", "")

body = "method=" + method + "\nquery=" + query + "\ncookie=" + cookie + "\n"

sys.stdout.write("Content-Type: text/plain\r\n")
sys.stdout.write("Content-Length: " + str(len(body)) + "\r\n")
sys.stdout.write("\r\n")
sys.stdout.write(body)
